intake/source/discovery.py

```diff
@@ -12,19 +12,158 @@
+        if not dotted_object_name:
+            logger.debug('Name %s is banned in config file', name)
+            if name in group:
+                entrypoint = group[name]
+                del group[name]
+                logger.debug("Disabled entrypoint '%s = %s.%s'",
+                             entrypoint.name,
+                             entrypoint.module_name,
+                             entrypoint.object_name)
+            if name in package_scan_results:
+                cls = package_scan_results[name]
+                del package_scan_results[name]
+                logger.debug("Disabled package_scan result '%s = %s.%s'",
+                             name_, cls.__module__, cls.__name__)
+            continue
+        module_name, object_name = dotted_object_name.rsplit('.', 1)
+        entrypoint = entrypoints.EntryPoint(name, module_name, object_name)
+        logger.debug("Discovered config-specified '%s = %s.%s'",
+                     entrypoint.name,
+                     entrypoint.module_name,
+                     entrypoint.object_name)
+        if name in group:
+            shadowed = group[name]
+            logger.debug("Config shadowed entrypoint '%s = %s.%s'",
+                         shadowed.name,
+                         shadowed.module_name,
+                         shadowed.object_name)
+        if name in package_scan_results:
+            cls = package_scan_results[name]
+            del package_scan_results[name]
+            logger.debug("Config shadowed package scan result '%s = %s.%s'",
+                         name, cls.__module__, cls.__name__)
+        group[name] = entrypoint
 
-    This function searches path (or sys.path) for packages with names that
-    start with plugin_prefix.  Those modules will be imported and scanned for
-    subclasses of intake.source.base.Plugin.  Any subclasses found will be
-    instantiated and returned in a dictionary, with the plugin's name attribute
-    as the key.
+    # Discovery is complete.
+    
+    # Load entrypoints. Any that were shadowed or banned have already been
+    # removed above.
+    drivers = {}
+    for entrypoint in group.values():
+        try:
+            drivers[entrypoint.name] = _load_entrypoint(entrypoint)
+        except ConfigurationError:
+            logger.exception()
+            continue
+        logger.debug("Loaded entrypoint '%s = %s.%s'",
+                     entrypoint.name,
+                     entrypoint.module_name,
+                     entrypoint.object_name)
+
+    # TODO Un-comment this code to unleash warnings after most known intake_*
+    # packages have been updated to provide 'intake.drivers' entrypoints.
+    # if package_scan_results:
+    #     warnings.warn(
+    #         f"The drivers {package_scan_results} do not specify entry_points "
+    #         f"and were only discovered via a package scan. This may break in a "
+    #         f"future release of intake. The packages should be updated.")
```

Comment:
You could make this class a custom warning class that is filtered out by default, allowing people to turn it one via filterwarnings. Maybe a "FutureWarning" ?

Reply:
Great idea